force-app/main/default/classes/StaticTask.cls

```diff
@@ -9,28 +9,42 @@ public with sharing class StaticTask {
   . Then it parses data and return Map collection where key is meail and value is Contact
   Feel free to refactorthis class. Don't forget about single responsibility approach when you create a new method.
   */
-  public static Map<String, Contact> extractEmailToContactMap(String payload) {
-    Map<String, Contact> emailToContactMap = new Map<String, Contact>();
+
+  public static String extractor(String Str) {
+    String extract = str.split(':')[1].replaceAll(' ', '');
+    return extract;
+  }
+
+  public static List<Contact> listContact(String payload) {
```

Comment:
This method is still pretty complex and hard to read. It has 3 levels of nested statements: 
1. `for (String contactString : stringOfContacts) {`
2. `for (String con : contactAsList) {`
3. `if (con.contains...`

Can you try to split it more?